lib/new_relic/agent/configuration/default_source.rb

```diff
@@ -1094,6 +1094,14 @@ def self.enforce_fallback(allowed_values: nil, fallback: nil)
           :allowed_from_server => false,
           :description => 'Controls auto-instrumentation of gRPC clients at start up.  May be one of [auto|prepend|chain|disabled].'
         },
+        :'instrumentation.grpc.host_denylist' => {
+          :default => ['tracing\.(staging-)?edge\.nr-data'],
```

Comment:
Given that we are not intending to ever perform a capture with the regex, `(staging-)` should be `(?:staging)` to make our non-capturing intent clear to Ruby (and future human observers).

Reply:
resolved: b05ee085cfd6e726be0285827c325be5a5aca9cd